doc/source/user/basics.types.rst

```diff
@@ -12,87 +12,124 @@ Array types and conversions between types
 NumPy supports a much greater variety of numerical types than Python does.
 This section shows which are available, and how to modify an array's data-type.
 
-The primitive types supported are tied closely to those in C:
+NumPy provides both bit sized type names and names based on the names of C types.
+Since the definition of C types are platform dependent, this means the explicitly
+bit sized should be preferred to avoid platform-dependent behavior in programs
+using NumPy.
+
+To ease integration with C code, where it is more natural to refer to
+platform-dependent C types, NumPy also provides type aliases that correspond
+to the C types for the platform. Some dtypes have trailing underscore to avoid
+confusion with builtin python type names, such as `numpy.bool_`.
 
 .. list-table::
     :header-rows: 1
 
-    * - Numpy type
-      - C type
+    * - Canonical Python API name
+      - Python API "C-like" name
+      - Actual C type
       - Description
 
     * - `numpy.bool_`
-      - ``bool``
-      - Boolean (True or False) stored as a byte
+      - N/A
+      - ``stdbool``
```

Comment:
just `bool`, you could note it's defined in `stdbool.h` though. You could also note that `intptr_t` and `uintptr_t` are in `stdint.h` below.

Reply:
Sure! I updated these rows.